R/broom_wrapper.R

```diff
@@ -879,12 +893,15 @@ prediction_survfit <- function(df, newdata = NULL, ...){
     # convert characterized data back to numeric.
     ret <- ret %>% mutate(estimate = as.numeric(estimate), std_error = as.numeric(std_error), conf_high = as.numeric(conf_high), conf_low = as.numeric(conf_low))
     # replace the cohort name with a string that is a concatenation of values that represents the cohort.
-    cohorts_labels <- newdata %>% unite(label, everything())
-    for (i in 1:nrow(newdata)){
+    # but, omit newdata column that has only 1 unique value from the cohort names we create here.
+    selected_newdata_colnames <- non_single_value_colnames(newdata)
+    selected_newdata <- newdata[, selected_newdata_colnames]
```

Comment:
Can this be length 0? If so, 1:nrow(selected_newdata) in l900 might cause an error